### PR TITLE
Add loggin to application insights

### PR DESCRIPTION
**Pull request by moumot**

### CODE REVIEW

File: SamMou.Api/Controllers/WeatherForecastController.cs [hunk 1]
```diff
@@ -27,6 +27,7 @@ public async Task<IActionResult> GetAll()
         {
             try
             {
+                _logger.LogInformation("Called GetAll");
```
Comment (samueljklee) [inline SamMou.Api/Controllers/WeatherForecastController.cs:30]: Do you have example output? It'll show the other metadata such as which method called this, right?

File: SamMou.Api/appsettings.Development.json [hunk 2]
```diff
@@ -10,5 +10,6 @@
     "Endpoint": "https://sammoucosmos.documents.azure.com:443/",
     "PrimaryKey": "",
     "DatabaseName": "SamMou"
-  }
+  },
+  "APPINSIGHTS_INSTRUMENTATIONKEY": ""
```
Comment (samueljklee) [inline SamMou.Api/appsettings.Development.json:14]: different format than other constants?

File: SamMou.Api/Program.cs [hunk 3]
```diff
@@ -1,26 +1,62 @@
...
+            var loggerConfiguration = new LoggerConfiguration();
+            loggerConfiguration.WriteTo
+                .ApplicationInsights(telemetryConfiguration,
+                    TelemetryConverter.Traces);
+
+            Log.Logger = loggerConfiguration
+                .Enrich.FromLogContext()
+                .CreateLogger();
+
+
```
Comment (samueljklee) [inline SamMou.Api/Program.cs:38]: nit
```suggestion

```

**Review by samueljklee — APPROVED** [review]

File: SamMou.Api/Controllers/WeatherForecastController.cs (hunk 1 above)
Comment (moumot) [reply SamMou.Api/Controllers/WeatherForecastController.cs:30]: yup you can go to application insights to see the output

Comment (moumot) [reply SamMou.Api/Controllers/WeatherForecastController.cs:30]: image

File: SamMou.Api/appsettings.Development.json (hunk 2 above)
Comment (moumot) [reply SamMou.Api/appsettings.Development.json:14]: cause this one only single string not a structure

File: SamMou.Api/Controllers/WeatherForecastController.cs (hunk 1 above)
Comment (samueljklee) [reply SamMou.Api/Controllers/WeatherForecastController.cs:30]: Saw it! What's the retention policy for our logs?

File: SamMou.Api/appsettings.Development.json (hunk 2 above)
Comment (samueljklee) [reply SamMou.Api/appsettings.Development.json:14]: okay

**Pull request merged**